Public/Src/Engine/Processes/SandboxedProcessInfo.cs

```diff
@@ -101,11 +102,17 @@ public sealed class SandboxedProcessInfo
             bool testRetries = false,
             LoggingContext loggingContext = null,
             IDetoursEventListener detoursEventListener = null,
-            IKextConnection sandboxedKextConnection = null)
+            IKextConnection sandboxedKextConnection = null,
+            AnyBuildShimInfo shimInfo = null)

```

Comment:
>AnyBuildShimInfo shimInfo = null [](start = 12, length = 32)

Why do we need to pass `shimInfo` to the constructor? I don't see it being used in this class besides assigning it to `fileAccessManifest`, which I think is better to do that outside of this class.

Reply:
Purely convenience - the calling code in another repo looks like:

            var shimInfo = new AnyBuildShimInfo(AbsolutePath.Create(_pathTable, shimPath));

            var sandboxedProcessInfo = new SandboxedProcessInfo(
                _pathTable,
                new LocalSandboxedFileStorage(),
                executablePath,
                disableConHostSharing: true,
                loggingContext: loggingContext,
                shimInfo: shimInfo)

The ctor being used implicitly creates a FileManifest, so wiring up this way allowed that shorter constructor to work properly.
I can remove this approach... what is your preference?


---
In reply to: [282687412](https://github.com/microsoft/BuildXL/pull/279#discussion_r282687412) [](ancestors = 282687412)

Reply:
My preference is not to modify `FileAccessManifest` inside `SandboxedProcessInfo`. For the case you mentioned above, you can always modify it to
```
          var shimInfo = new AnyBuildShimInfo(AbsolutePath.Create(_pathTable, shimPath));
          var fam = new FileAccessManifest(pathTable);
          fam.AnyBuildShimInfo = shimInfo;
           var sandboxedProcessInfo = new SandboxedProcessInfo(
                _pathTable,
                new LocalSandboxedFileStorage(),
                executablePath,
                disableConHostSharing: true,
                loggingContext: loggingContext,
                shimInfo: shimInfo)
```

---
In reply to: [282945198](https://github.com/microsoft/BuildXL/pull/279#discussion_r282945198) [](ancestors = 282945198,282687412)

Reply:
Removed.